### PR TITLE
Update Azure.AppService.Middleware to 1.5.2

### PR DESCRIPTION
**Pull request by karshinlin**
### Issue describing the changes in this PR

Updating the Auth middleware for Linux Consumption.
Packages are Azure.AppService.Middleware* -> v1.5.2

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj [hunk 1]
```diff
@@ -10,7 +10,7 @@
     <AspNetCoreHostingModel>InProcess</AspNetCoreHostingModel>
     <TieredCompilation>false</TieredCompilation>
     <NoWarn>NU5104</NoWarn>
-    <IdentityDependencyVersion>6.31.0</IdentityDependencyVersion>
```
Comment (karshinlin) [inline src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj]: cc @fabiocav @brettsam @mathewc 

This breaks dependency test with the following output. Please confirm we are good to update deps.json
```
Changed:
        - Microsoft.IdentityModel.Abstractions.dll: 6.31.0.0/6.31.0.40607 -> 6.32.0.0/6.32.0.40712
        - Microsoft.IdentityModel.JsonWebTokens.dll: 6.31.0.0/6.31.0.40607 -> 6.32.0.0/6.32.0.40712
        - Microsoft.IdentityModel.Logging.dll: 6.31.0.0/6.31.0.40607 -> 6.32.0.0/6.32.0.40712
        - Microsoft.IdentityModel.Protocols.dll: 6.10.0.0/6.10.0.20330 -> 6.32.0.0/6.32.0.40712
        - Microsoft.IdentityModel.Protocols.OpenIdConnect.dll: 6.10.0.0/6.10.0.20330 -> 6.32.0.0/6.32.0.40712
        - Microsoft.IdentityModel.Tokens.dll: 6.31.0.0/6.31.0.40607 -> 6.32.0.0/6.32.0.40712
        - System.IdentityModel.Tokens.Jwt.dll: 6.31.0.0/6.31.0.40607 -> 6.32.0.0/6.32.0.40712
    
Removed:
    
Added:
        - Microsoft.IdentityModel.Validators.dll: 6.32.0.0/6.32.0.40712
 ```

File: src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj [hunk 2]
```diff
@@ -72,7 +72,7 @@
     <PackageReference Include="Microsoft.AspNet.WebApi.Client" Version="5.2.8" />
     <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="6.0.0" />
     <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="6.0.0" />
-    <PackageReference Include="Microsoft.Azure.AppService.Middleware.Functions" Version="1.5.1" />
+    <PackageReference Include="Microsoft.Azure.AppService.Middleware.Functions" Version="1.5.2-beta.2" />
```
Comment (karshinlin) [inline src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj]: Using beta version while validating end to end. Will publish 1.5.2 after all validation succeeds and before merging PR.

File: src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj (hunk 1 above)
Comment (brettsam) [reply src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj]: hey @karshinlin -- can you try rebasing on dev? @fabiocav should have updated these and taken care of the test already in https://github.com/Azure/azure-functions-host/pull/9409

Comment (karshinlin) [reply src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj]: @brettsam latest failure output. Can I add this new dependency? (This is needed as part of the package update for sure)

      Changed:

  Removed:

  Added:
    - Microsoft.IdentityModel.Validators.dll: 6.32.0.0/6.32.0.40712

Comment (brettsam) [reply src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj]: You'll need to add this to the "runtimeassemblies.json" file and mark it as "private". There should be other examples in there of private assemblies (should just be a few lines of json). Just make sure to put it in alphabetical order with other assemblies.

(https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/runtimeassemblies.json)

**Review by yojagad — APPROVED** [review]

**Review by pragnagopa — APPROVED** [review]

**Review by brettsam — APPROVED** [review]

**Pull request merged**